### ISSUE TITLE
updatingRREFdict errors

### ISSUE THREAD

**Issue by wortelus** [leleopard/pyXPUDPServer#4]
My software for datarefs is running in loop, and every once and then it runs into this error
```  
File "C:\Users\wortelus\AppData\Local\Programs\Python\Python310\lib\threading.py", line 1009, in _bootstrap_inner
    self.run()
  File "C:\Users\wortelus\PycharmProjects\xpaneler\venv\lib\site-packages\pyxpudpserver\XPlaneUDPServer.py", line 540, in run
    for index, RREF in self.datarefsIndices.items():
RuntimeError: dictionary changed size during iteration
```

I think it is caused by setting the updatingRREFdict true while the if condition has already been passed, and the change is then executed during the loop.

Shouldn't be locking utilized instead?

**Comment by wortelus** [comment]
Got a personal solution.. created a pull request for this #5

**Comment by wortelus** [comment]
Fixed in #5
